java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/PeriodicEmitLastValueOperator.java

```diff
@@ -0,0 +1,81 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.udf;
+
+import org.apache.flink.api.common.operators.ProcessingTimeService;
+import org.apache.flink.api.common.state.ListState;
+import org.apache.flink.api.common.state.ListStateDescriptor;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.types.Row;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Iterator;
+
+/** An operator that periodically emits the last value it has received from upstream operator. */
+public class PeriodicEmitLastValueOperator extends AbstractStreamOperator<Row>
+        implements OneInputStreamOperator<Row, Row>, Serializable, BoundedOneInput {
+    private final long emitIntervalMs;
+
+    private transient ListState<Row> state;
+
+    public PeriodicEmitLastValueOperator(Integer emitIntervalMs) {
+        this.emitIntervalMs = emitIntervalMs.longValue();
+    }
+
+    @Override
+    public void initializeState(StateInitializationContext context) throws Exception {
+        super.initializeState(context);
+        state =
+                context.getOperatorStateStore()
+                        .getListState(new ListStateDescriptor<>("lastRowState", Row.class));
+    }
+
+    @Override
+    public void processElement(StreamRecord<Row> element) throws Exception {
+        if (!state.get().iterator().hasNext()) {
+            getProcessingTimeService()
+                    .registerTimer(
+                            getProcessingTimeService().getCurrentProcessingTime() + emitIntervalMs,
+                            new EmitLastValueCallback());
+        }
+        state.update(Collections.singletonList(element.getValue()));
+    }
+
+    @Override
+    public void endInput() throws Exception {
+        Iterator<Row> iterator = state.get().iterator();
+        if (iterator.hasNext()) {
+            output.collect(new StreamRecord<>(iterator.next()));
+        }
+    }
+
+    private class EmitLastValueCallback implements ProcessingTimeService.ProcessingTimeCallback {
+        @Override
+        public void onProcessingTime(long l) throws Exception {
+            output.collect(new StreamRecord<>(state.get().iterator().next()));
+            getProcessingTimeService()
+                    .registerTimer(
+                            getProcessingTimeService().getCurrentProcessingTime() + emitIntervalMs,
```

Comment:
I think we can use is passed as parameter `l` to register the next processing time timer.